src/web_interface/rest/rest_firmware.py

```diff
@@ -120,23 +139,45 @@ def _get_with_uid(self, uid):
         fitted_firmware = self._fit_firmware(firmware)
         return success_message(dict(firmware=fitted_firmware), self.URL, request_data=dict(uid=uid))
 
-    def _put_without_uid(self):
-        try:
-            data = convert_rest_request(request.data)
-        except TypeError as type_error:
-            return error_message(str(type_error), self.URL, request_data=request.data)
-
-        result = self._process_data(data)
-        if 'error_message' in result:
-            logging.warning('Submission not according to API guidelines! (data could not be parsed)')
-            return error_message(result['error_message'], self.URL, request_data=data)
-
-        logging.debug('Upload Successful!')
-        return success_message(result, self.URL, request_data=data)
+    @staticmethod
+    def _fit_firmware(firmware):
+        meta = create_meta_dict(firmware)
+        analysis = firmware.processed_analysis
+        return dict(meta_data=meta, analysis=analysis)
 
-    def _put_with_uid(self, uid):
+    @roles_accepted(*PRIVILEGES['submit_analysis'])
+    @api.expect(firmware_model)
+    def put(self, uid):
+        '''
+        Upload a firmware
+        The HTTP body shall contain a json document of the structure shown below
```

Comment:
shall -> must
shall sounds very ... archaic (I know it is like this in the REST docs, too. We really need to revise them. Maybe you can update it there, too)